src/float.cr

```diff
@@ -142,16 +142,23 @@ struct Float32
     value.to_f32 whitespace: whitespace, strict: strict
   end
 
-  # Returns a `Float32` by invoking `to_f32` on *value*.
-  def self.new(value)
+  # Casts the *value* by invoking `to_f32` on it.
+  def self.new(value) : self
     value.to_f32
   end
 
-  # Returns a `Float32` by invoking `to_f32!` on *value*.
+  # Casts the *value* by invoking `to_f32!` on it.
   def self.new!(value) : self
     value.to_f32!
   end
 
+  # Casts the *value* by invoking `to_f32?` on it.
+  def self.new?(value) : self?
+    # TODO: implement to_f32?
+    value.to_f32
```

Comment:
`Float32.to_f32?` is not yet implemented.
Yes it could be done here, or another PR probably.

Reply:
`Float32#to_f32?` and other similar methods are not yet implemented.
Yes it could be done here, or another PR probably.

Reply:
At the very least, these methods should call the questionmark methods when the receiver responds to them.
It's probably fine to call the methods without questionmark as a fallback. Or it might actually be better to leave the unimplemented cases out for now.